test/check/check.go

```diff
@@ -389,6 +394,10 @@ func checkForComponentRemoval(component *Component) error {
 		errsAppend(checkForPrometheusRuleRemoval(component.PrometheusRule))
 	}
 
+	if component.NetworkAttachmentDefinition != "" {
+		errsAppend(checkForNetworkAttachmentDefinitionRemoval(component.NetworkAttachmentDefinition))
+	}
+
```

Comment:
The condition can be moved into the function
but we can consider doing it on a follow-up for all those methods not in this PR

Reply:
>but we can consider doing it on a follow-up for all those methods not in this PR
+!